Makefile

```
@@ -39,43 +53,49 @@ $(MOCK_DIR)/client.go: $(GO_BIN)/mockgen $(MOCK_SRC_DIR)
 #-----------------------------------------------------------------------------------------------------------------------
 # Phony Rules(https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html)
 #-----------------------------------------------------------------------------------------------------------------------
-
 .PHONY: build run clean test lint audit format
 
-build:
+help: ## Show this help
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+build: ## Build the CLI binary
 	@echo "==> Building binary within ${BUILD_DIR}/${BINARY_NAME}"
-	@go build -v -o ${BUILD_DIR}/${BINARY_NAME} main.go
+	@go build -v -ldflags "$(GO_LINKER_FLAGS)" -o "${BUILD_DIR}/${BINARY_NAME}" "$(CURDIR)/cmd/fga/main.go"
  
-run: build
-	${BUILD_DIR}/${BINARY_NAME} $(ARGS)
+install: ## Install the CLI binary
+	@$(MAKE) build BUILD_DIR="$(GO_BIN)"
+
+run: $(GO_BIN)/CompileDaemon ## Watch for changes and recompile the CLI binary
+	@echo "==> Watching for changes"
+	@CompileDaemon -build='make install' -command='fga --version'
  
-clean:
+clean: ## Clean project files
 	@echo "==> Cleaning project files"
-	go clean
-	rm -f ${BUILD_DIR}
+	@go clean
+	@rm -f ${BUILD_DIR}
 
-test:
+test: ## Run tests
 	go test -race \
-			-coverpkg=./... \
-			-coverprofile=coverageunit.tmp.out \
-			-covermode=atomic \
-			-count=1 \
-			-timeout=5m \
-			./...
+		-coverpkg=./... \
+		-coverprofile=coverageunit.tmp.out \
+		-covermode=atomic \
+		-count=1 \
+		-timeout=5m \
+		./...
 	@cat coverageunit.tmp.out | grep -v "mocks" > coverageunit.out
 	@rm coverageunit.tmp.out
 
-lint: $(GO_BIN)/golangci-lint
+lint: $(GO_BIN)/golangci-lint ## Lint Go source files
 	@echo "==> Linting Go source files"
 	@golangci-lint run -v --fix -c .golangci.yaml ./...
 
-audit: $(GO_BIN)/govulncheck
+audit: $(GO_BIN)/govulncheck ## Audit Go source files
 	@echo "==> Checking Go source files for vulnerabilities"
 	govulncheck ./...
 
-format: $(GO_BIN)/gofumpt $(GO_BIN)/gci
+format: $(GO_BIN)/gofumpt $(GO_BIN)/gci ## Format Go source files
```

I think we removed `gci`, so can we remove it here?
```suggestion
format: $(GO_BIN)/gofumpt ## Format Go source files
```